src/sentry/snuba/metrics/extraction.py

```diff
@@ -1430,6 +1444,27 @@ def _parse_query(value: str) -> QueryParsingResult:
             raise Exception(f"Invalid search query '{value}' in on demand spec: {e}")
 
 
+def fetch_on_demand_metric_spec(
```

Comment:
This can now be used by the tests to have a single entry point to querying the right spec based on feature flags.

Jump to the end of `tests/sentry/relay/config/test_metric_extraction.py` to see.